pkgs/development/tools/build-managers/waf/setup-hook.sh

```diff
@@ -1,20 +1,40 @@
 wafConfigurePhase() {
     runHook preConfigure
 
-    if ! [ -f ./waf ]; then
-        cp @waf@ waf
+    if ! [ -f "${wafPath:=./waf}" ]; then
+        echo "copying waf to $wafPath..."
+        cp @waf@ "$wafPath"
     fi
 
     if [[ -z "${dontAddPrefix:-}" && -n "$prefix" ]]; then
         configureFlags="${prefixKey:---prefix=}$prefix $configureFlags"
     fi
 
-    local flagsArray=(
-        $configureFlags ${configureFlagsArray[@]}
+    local flagsArray=(@crossFlags@);
+    for flag in $configureFlags "${configureFlagsArray[@]}";
+    do
+        if [[
+        # waf does not support these flags, but they are "blindly" added by the
+        # pkgsStatic overlay, for example.
+              $flag != "--enable-static"
```

Comment:
Perhaps we should have some separate variable like "wafFlags" for this? waf supports a subset of the configure flags but not enough apparently.

Reply:
This would be a more invasive change. Would you prefer I change the PR to do this?

Reply:
It's okay as is for now. This is more a note on this.